File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Projects/Commands/StartWorkItemCommand.cs
using SIE.EMS.EarlierStage.Projects;
using SIE.Web.Command;

namespace SIE.Web.EMS.EarlierStage.Projects.Commands
{
    /// <summary>
    /// 计划开始
    /// </summary>
    [JsCommand("SIE.Web.EMS.EarlierStage.Projects.Commands.StartWorkItemCommand")]
    public class StartWorkItemCommand : ViewCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var id = args.Data.ToJsonObject<double>();
            RT.Service.Resolve<ProjectController>().StartWorkItem(id);
            return true;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.EarlierStage/Projects/Commands/StartWorkItemCommand.js
SIE.defineCommand('SIE.Web.EMS.EarlierStage.Projects.Commands.StartWorkItemCommand', {
    meta: { text: "开始", group: "edit", iconCls: "icon-Play icon-blue" },
    canExecute: function (view) {
        if (view.getSelection() == null || view.getSelection().length !== 1) {
            return false;
        }
        var p = view.getCurrent();
        if (p == null) return false;
        if (view.getParent().getSelection().first().getProjectStatus() == 30) return false;
        if (p.data.WorkStatus !== 10) return false;
        return true;
    },
    execute: function (view, source) {
        var p = view.getCurrent();
        view.execute({
            data: p.getId(),
            success: function (res) {
                SIE.Msg.showMessage("提交成功!".t());
                view.reloadData();
            }
        });
    }
});